HopsanGenerator/src/generators/HopsanFMIGenerator.cpp

```diff
@@ -1618,6 +1643,29 @@ bool HopsanFMIGenerator::generateModelDescriptionXmlFile(ComponentSystem *pSyste
         }
     }
 
+    QList<ParameterSpecification> parameterSpecs;
+    getParameters(parameterSpecs, pSystem);
+
+    for(const auto &parSpec : parameterSpecs)
+    {
+        QDomElement varElement = domDocument.createElement("ScalarVariable");
+        varElement.setAttribute("name", parSpec.name);
+        varElement.setAttribute("valueReference", (unsigned int)vr);
+        varElement.setAttribute("causality", "parameter");
+        varElement.setAttribute("initial","exact");
+        varElement.setAttribute("variability", "fixed");
+        varElement.setAttribute("description", parSpec.description);
+        QDomElement dataElement = domDocument.createElement(parSpec.type);
+        dataElement.setAttribute("start", parSpec.init);   //! @todo Support start values
```

Comment:
I found the code I wrote, and I also appended the attribute "unit", I think that in this case it should be the "base unit" typically SI unit if a unit has been set for the parameter

Reply:
I added support for units, and confirmed that the units show up when importing the FMU back to Hopsan.